### DISCUSSION TITLE
Are there any ways to get the boolean value of screen.find?

### DISCUSSION THREAD

**Discussion by louisho5** [nut-tree/nut.js#339] [Q&A] [answered]
For example, I want a mouse move action based on the found image
`````
if ( screen.find("image.png")) ) {
    mouse.move(straightTo(centerOf(screen.find("image.png"))));
} else {
    mouse.move(down(10));
}
`````
Apparently, the above code doesn't work. I am wondering if there is any way may achieve what I expected? Thanks everyone.

**Comment by s1hofmann** [comment, marked as answer]
Late reply here, since I didn't check discussions regularly.

One way to achieve your goal would be to use a `catch` block.
In case the image is not found, the `find` Promise will reject, thus you could move the mouse conditionally in this case.

Something like

```js
async function demo() {
    try {
        const targetRegion = await screen.find("image.png");
        await mouse.move(straightTo(centerOf(targetRegion)));
    } catch (e) {
        await mouse.move(down(10));
    }
}
```

**Reply by louisho5** [reply]
Thanks @s1hofmann, the answer is clear! 
So what if I want to add more variations, may I use the nested try-catch to achieve?

Like below
```js
async function demo() {
    try {
        const target1 = await screen.find("image1.png");
        await mouse.move(straightTo(centerOf(target1)));
    } catch (e) {
        try {
          const target2 = await screen.find("image2.png");
          await mouse.move(straightTo(centerOf(target2)));
      } catch (e) {
          await mouse.move(down(10));
      }
    }
}
```

**Reply by s1hofmann** [reply]
That would be one option, yes
